internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,120 @@
+func (c *V2HttpController) AddUpdateDevice(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var device contract.Device
+	err := json.NewDecoder(request.Body).Decode(&device)
+	if err != nil {
+		c.sendError(writer, request, edgexErr.KindServerError, "failed to decode JSON", err, sdkCommon.APIV2DeviceCallbackRoute, correlationID)
+		return
+	}
+
+	isAdd := request.Method == http.MethodPost
+	edgexErr := application.AddUpdateDevice(isAdd, device, c.dic)
+	if edgexErr == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, edgexErr, sdkCommon.APIV2DeviceCallbackRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) DeleteProfile(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteProfile(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2ProfileCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2ProfileCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateProfile(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var profile contract.DeviceProfile
+	err := json.NewDecoder(request.Body).Decode(&profile)
+	if err != nil {
+		c.sendError(writer, request, edgexErr.KindServerError, "failed to decode JSON", err, sdkCommon.APIV2ProfileCallbackRoute, correlationID)
+		return
+	}
+
+	isAdd := request.Method == http.MethodPost
+	edgexErr := application.AddUpdateProfile(isAdd, profile, c.dic)
+	if edgexErr == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2ProfileCallbackRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, edgexErr, sdkCommon.APIV2ProfileCallbackRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) DeleteWatcher(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteWatcher(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2WatcherCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2WatcherCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateWatcher(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var watcher contract.ProvisionWatcher
```

Comment:
Yup, not created yet by Core Service, but it will have the same separate Add and Update request pattern. I think it should be add now before this PR can be completed.

Reply:
@lenny-intel we missed the provision watcher in swagger, dto, and API implementation.  I thought we should remove this part in this PR.  If you agree, @hahattan could open an issue for "V2 Provision Watcher callback" with label "Ireland".

Reply:
Sound good!